lib/compress/zstd_compress.c

```diff
@@ -4484,6 +4489,27 @@ static void ZSTD_overflowCorrectIfNeeded(ZSTD_matchState_t* ms,
     }
 }
 
+#include "zstd_preSplit.h"
+
+static size_t ZSTD_optimalBlockSize(ZSTD_CCtx* cctx, const void* src, size_t srcSize, size_t blockSizeMax, ZSTD_strategy strat, S64 savings)
+{
+    /* note: conservatively only split full blocks (128 KB) currently,
+     * and even then only if there is more than 128 KB input remaining.
+     */
+    if (srcSize <= 128 KB || blockSizeMax < 128 KB)
```

Comment:
Does this mean that the block splitter won't work in streaming mode? What if the user passes in exactly 128KB at a time?

Reply:
Indeed, if the user passes exactly 128 KB at a time, block splitting will not be triggered.

Reply:
I don't think that is a good idea. In my model of the streaming API, how you chunk your data doesn't impact the compression, with the exception of the final block. This seems like it could introduce confusing behavior. Can we get rid of this restriction?

Reply:
OK, I'll look into it.

Reply:
All full blocks are now sent to the splitter function, irrespective of the presence of additional data beyond the full block.